decision_service/dsclientlib/logger/ds_concurrent_queue.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include <queue>
+#include <mutex>
+
+
+namespace decision_service {
+
+	//a concurrent queue with locks and mutex
+	template <class T>
+	class concurrent_queue {
```

Comment:
yes.

---
In reply to: [187105931](https://github.com/JohnLangford/vowpal_wabbit/pull/1479#discussion_r187105931) [](ancestors = 187105931)

Reply:
It seems ok to use then.  I'm skeptical we'll want to use it in the long term since we'd like to shift towards extents on a ring buffer.  In the short term, locking is a relatively expensive operation so eliminating that could be desirable.

Reply:
test

Reply:
test 2

---
In reply to: [187641806](https://github.com/JohnLangford/vowpal_wabbit/pull/1479#discussion_r187641806) [](ancestors = 187641806)